csharp/ql/src/Bad Practices/Control-Flow/ConstantCondition.ql

```diff
@@ -71,8 +71,10 @@ class ConstantNullnessCondition extends ConstantCondition {
 
   ConstantNullnessCondition() {
     forex(ControlFlow::Node cfn | cfn = this.getAControlFlowNode() |
-      exists(ControlFlow::SuccessorTypes::NullnessSuccessor t | exists(cfn.getASuccessorByType(t)) |
-        b = t.getValue()
+      exists(ControlFlow::SuccessorTypes::NullnessSuccessor t, ControlFlow::Node s |
+        s = cfn.getASuccessorByType(t) |
+        b = t.getValue() and
+        not s.isJoin()
```

Comment:
This will also include cases such as `(a ? b : null)?.Foo`, but I guess that makes sense. Perhaps add a test case for that as well?